File: 2022/01/main.ts
import { readLines } from "https://deno.land/std/io/buffer.ts";

async function getInput() {
    const lines = [];
    for await (const line of readLines(Deno.stdin)) {
        lines.push(line);
    }

    return lines;
}

async function main() {
    const input = await getInput();
    const bags = [0];    

    for (const line of input) {
        switch (line) {
            case "":
                bags.push(0);
                break;
            default:
                bags[bags.length - 1] += Number.parseInt(line);
                break;
        }
    }

    console.log('Part 1:', bags
                            .reduce((a, b) => Math.max(a, b), -1)
    );

    console.log('Part 2:', bags
                            .sort()
                            .slice(0, 3)
                            .reduce((a, b) => a + b, 0)
    );
}

await main();

File: 2022/01/main.erl
-module(main).
-export([main/1]).
-import(lists,[reverse/1]). 

main(_) -> 
    Lines = read_stdin([]),
    Bags = list_to_bags(Lines, [0]),
    BiggestBag = lists:max(Bags),
    SumTop3 = lists:sum(lists:sublist(reverse(lists:sort(Bags)), 3)),
    io:fwrite(io:format("Part 1: ~p\nPart 2: ~p\n", [BiggestBag, SumTop3])).

list_to_bags([], Bags) ->
    Bags;

list_to_bags([Head|Tail], Bags) ->
    TrimmedHead = string:trim(Head),
    case TrimmedHead of
        "" ->
            list_to_bags(Tail, Bags ++ [0]);
        _ ->
            [BagTail|BagInitReversed] = reverse(Bags),
            {PartNumber, _} = string:to_integer(TrimmedHead),
            NewBagTail = BagTail + PartNumber,
            list_to_bags(Tail, reverse([NewBagTail|BagInitReversed]))
    end.

read_stdin(Acc) ->
    case io:get_line("") of
        eof ->
            Acc; 
        Line ->
            read_stdin(Acc ++ [Line])
    end.